manager/src/main/java/org/openremote/manager/mqtt/UserAssetProvisioningMQTTHandler.java

```diff
@@ -329,23 +329,43 @@ protected String getResponseTopic(Topic topic) {
+    protected void processX509ProvisioningMessage(RemotingConnection connection, Topic topic, ProvisioningMessage provisioningMessage) {
 
         LOG.fine(() -> "Processing X.509 provisioning message: " + MQTTBrokerService.connectionToString(connection));
 
-        if (TextUtil.isNullOrEmpty(provisioningMessage.getCert())) {
-            LOG.info("Certificate is missing from X509 provisioning message" + MQTTBrokerService.connectionToString(connection));
-            publishMessage(getResponseTopic(topic), new ErrorResponseMessage(ErrorResponseMessage.Error.CERTIFICATE_INVALID), MqttQoS.AT_MOST_ONCE);
-            return;
-        }
-
-        // Parse client cert
         X509Certificate clientCertificate;
-        try {
-            clientCertificate = ProvisioningUtil.getX509Certificate(provisioningMessage.getCert());
-        } catch (CertificateException e) {
-            LOG.log(Level.INFO, "Failed to parse X.509 certificate: " + MQTTBrokerService.connectionToString(connection), e);
-            publishMessage(getResponseTopic(topic), new ErrorResponseMessage(ErrorResponseMessage.Error.CERTIFICATE_INVALID), MqttQoS.AT_MOST_ONCE);
+
+        String certUniqueId;
+        String uniqueId;
+        boolean isMTLS = provisioningMessage instanceof MTLSProvisioningMessage;
+
+        if (provisioningMessage instanceof X509ProvisioningMessage) {
+            try {
+                clientCertificate = ProvisioningUtil.getX509Certificate(((X509ProvisioningMessage) provisioningMessage).getCert());
+            } catch (CertificateException e) {
+                LOG.log(Level.INFO, "Failed to parse X.509 certificate: " + MQTTBrokerService.connectionToString(connection), e);
+                publishMessage(getResponseTopic(topic), new ErrorResponseMessage(ErrorResponseMessage.Error.CERTIFICATE_INVALID), MqttQoS.AT_MOST_ONCE);
+                return;
+            }
+
+            certUniqueId = ProvisioningUtil.getSubjectCN(clientCertificate.getSubjectX500Principal());
+            uniqueId = topicTokenIndexToString(topic, 1);
+        } else if (provisioningMessage instanceof MTLSProvisioningMessage) {
+            clientCertificate = connection.getSubject().getPrivateCredentials(X509CertificateCredential.class).stream()
+                .findFirst()
+                .map(X509CertificateCredential::getCertificate)
+                .orElse(null);
+            if(clientCertificate == null){
+                LOG.info("No client certificate found in MTLS provisioning message: " + MQTTBrokerService.connectionToString(connection));
+                publishMessage(getResponseTopic(topic), new ErrorResponseMessage(ErrorResponseMessage.Error.CERTIFICATE_INVALID), MqttQoS.AT_MOST_ONCE);
+                return;
+            }
+
+            certUniqueId = ProvisioningUtil.getSubjectCN(clientCertificate.getSubjectX500Principal());
+            uniqueId = ProvisioningUtil.getSubjectCN(clientCertificate.getSubjectX500Principal());
```

Comment:
Need to still verify the topic index 1 string matches the unique ID if we are still using the same handler